config/config.go

```diff
@@ -0,0 +1,199 @@
+
+	yamlBytes, yamlErr := yaml.Marshal(&Config)
+	if yamlErr != nil {
+		return yamlErr
+	}
+
+	writeErr := ioutil.WriteFile(ConfigFilePath, yamlBytes, 0600)
+	if writeErr != nil {
+		return writeErr
+	}
+
+	return nil
+}
+
+// ReadFromFile reads configuration from the YAML config file
+func readFromFile(filePath string) (*configStruct, error) {
+	myConfig := new(configStruct)
+
+	data, readErr := ioutil.ReadFile(filePath)
+	if readErr != nil {
+		if os.IsNotExist(readErr) {
+			// ignore if file does not exist,
+			// as this is not an error.
+			return myConfig, nil
+		}
+		return nil, readErr
+	}
+
+	yamlErr := yaml.Unmarshal(data, &myConfig)
+	if yamlErr != nil {
+		return nil, yamlErr
+	}
+
+	return myConfig, nil
+}
+
+// getKubeconfigPaths returns a slice of paths to known kubeconfig files
+func getKubeconfigPaths(homeDir string) []string {
+
+	// check if KUBECONFIG environment variable is set
+	kubeconfigEnv := os.Getenv("KUBECONFIG")
+	if kubeconfigEnv != "" {
+		// KUBECONFIG is set.
+		// Now check all the paths included for file existence
+
+		// "windows" is really the value for the windows platform.
+		// No constant for that in runtime.
+		pathSep := ":"
+		if runtime.GOOS == "windows" {
+			pathSep = ";"
+		}
+		paths := strings.Split(kubeconfigEnv, pathSep)
+		out := []string{}
+		for _, myPath := range paths {
+			if _, err := os.Stat(myPath); err == nil {
+				out = append(out, myPath)
+			}
+		}
+		return out
+	}
+
+	// KUBECONFIG is not set.
+	// Look for the default location ~/.kube/config
+	filePath := path.Join(homeDir, ".kube", "config")
+
+	if _, err := os.Stat(filePath); err == nil {
+		// file exists in default location
+		return []string{filePath}
+	}
+
+	// No kubeconfig file. Return empty slice.
+	return nil
+}
+
+// GetDefaultCluster determines which is the default cluster. This can be either
+// the only cluster accessible, or a cluster selected explicitly.
+func GetDefaultCluster() (clusterID string, err error) {
+	// Check selected cluster
+	if Config.Cluster != "" {
+		return Config.Cluster, nil
+	}
+	// Go through available orgs and clusters to find all clusters
+	if Config.Token == "" {
+		return "", errors.New("User not logged in.")
+	}
+	client := gsclientgen.NewDefaultApi()
+	authHeader := "giantswarm " + Config.Token
+	orgsResponse, _, err := client.GetUserOrganizations(authHeader)
+	if err != nil {
+		return "", err
+	}
+	if orgsResponse.StatusCode == 10000 {
+		if len(orgsResponse.Data) > 0 {
+			clusterIDs := []string{}
+			for _, orgName := range orgsResponse.Data {
+				clustersResponse, _, err := client.GetOrganizationClusters(authHeader, orgName)
+				if err != nil {
+					return "", err
+				}
+				for _, cluster := range clustersResponse.Data.Clusters {
+					clusterIDs = append(clusterIDs, cluster.Id)
+				}
+			}
+			if len(clusterIDs) == 1 {
+				return clusterIDs[0], nil
+			}
+			return "", nil
+		}
+	}
+	return "", errors.New(orgsResponse.StatusText)
```

Comment:
A general note on error handling. We use https://github.com/juju/errgo in most places which is useful when something goes wrong. In your current code you do not know where the code flows along when something breaks. Then debugging is a huge pain, especially for people who are not familiar with the code base. The approach that proved quite good was to have a `error.go` in each package that has to deal with errors. See e.g. https://github.com/giantswarm/kubernetesd/blob/master/service/creator/error.go. Errors being defined there can then be used like in the following places. 

- https://github.com/giantswarm/kubernetesd/blob/master/service/creator/creator.go#L76
- https://github.com/giantswarm/kubernetesd/blob/master/service/creator/creator.go#L106

You may want to create an issue for this.